packages/sysutils/busybox/scripts/functions

```diff
@@ -0,0 +1,166 @@
+################################################################################
+#      This file is part of LibreELEC - http://www.libreelec.tv
+#      Copyright (C) 2016 Neil MacLeod (milhouse@kodi.tv)
+#
+#  LibreELEC is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  LibreELEC is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with LibreELEC.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+# Show frame-buffer cursor
+showcursor() {
+  echo -en "\033[?25h"
+
+  # show cursor
+  if [ -f /sys/devices/virtual/graphics/fbcon/cursor_blink ] ; then
+    echo 1 > /sys/devices/virtual/graphics/fbcon/cursor_blink
+  fi
+}
+
+# Hide frame-buffer cursor
+hidecursor() {
+  echo -en "\033[?25l"
+
+  if [ -f /sys/devices/virtual/graphics/fbcon/cursor_blink ] ; then
+    echo 0 > /sys/devices/virtual/graphics/fbcon/cursor_blink
+  fi
+}
+
+# Start a progress meter
+# $1: spinner, percent, countdown
+# $2: message to display
+# [spinner]
+#   $3: optional command to execute
+#   $4: optional success completion message (nothing if not specified)
+#   $5: optional failure completion message ($4 if not specified)
+# [percent]
+#   $3: name of filename to be progress monitored
+#   $4: terminal size (in bytes) of $3 when progress is complete
+#   $5: optional command to execute
+#   $6: optional success message (100% if not specified)
+#   $7: optional failure message ($6 if not specified)
+# [countdown]
+#   $3: number of seconds to start counting down from
+#   $4: optional completion message, default is nothing
+StartProgress() {
+  local cmdresult=0
+
+  # Use files for inter-process communication.
+  # This file is used to indicate a metter is running/active
+  echo > /tmp/progress.run
+
+  case "$1" in
+    spinner)
+      ProgressTask_Spinner "$2" &
+      if [ -n "$3" ]; then
+        eval "$3"
+        cmdresult=$?
+        [ $cmdresult -eq 0 ] && StopProgress "${4}" || StopProgress "${5:-$4}"
```

Comment:
The way the progress meter is now written you can have it do everything for you if the command is relatively simple, or for more complex commands (or coding clarity) you can control the start and stop separately.